RecoBTag/FeatureTools/plugins/DeepBoostedJetTagInfoProducer.cc

```diff
@@ -405,26 +423,14 @@ void DeepBoostedJetTagInfoProducer::fillParticleFeatures(DeepBoostedJetFeatures
 
       TrackInfoBuilder trkinfo(track_builder_);
       trkinfo.buildTrackInfo(&(*cand), jet_dir, jet_ref_track_dir, *pv_);
-      if (use_puppiP4_) {
-        // to preserve the old behavior for DeepBoostedJet
-        fts.fill("pfcand_btagEtaRel", trkinfo.getTrackEtaRel());
-        fts.fill("pfcand_btagPtRatio", trkinfo.getTrackPtRatio());
-        fts.fill("pfcand_btagPParRatio", trkinfo.getTrackPParRatio());
-        fts.fill("pfcand_btagSip2dVal", trkinfo.getTrackSip2dVal());
-        fts.fill("pfcand_btagSip2dSig", trkinfo.getTrackSip2dSig());
-        fts.fill("pfcand_btagSip3dVal", trkinfo.getTrackSip3dVal());
-        fts.fill("pfcand_btagSip3dSig", trkinfo.getTrackSip3dSig());
-        fts.fill("pfcand_btagJetDistVal", trkinfo.getTrackJetDistVal());
-      } else {
-        fts.fill("pfcand_btagEtaRel", catch_infs(trkinfo.getTrackEtaRel()));
-        fts.fill("pfcand_btagPtRatio", catch_infs(trkinfo.getTrackPtRatio()));
-        fts.fill("pfcand_btagPParRatio", catch_infs(trkinfo.getTrackPParRatio()));
-        fts.fill("pfcand_btagSip2dVal", catch_infs(trkinfo.getTrackSip2dVal()));
-        fts.fill("pfcand_btagSip2dSig", catch_infs(trkinfo.getTrackSip2dSig()));
-        fts.fill("pfcand_btagSip3dVal", catch_infs(trkinfo.getTrackSip3dVal()));
-        fts.fill("pfcand_btagSip3dSig", catch_infs(trkinfo.getTrackSip3dSig()));
-        fts.fill("pfcand_btagJetDistVal", catch_infs(trkinfo.getTrackJetDistVal()));
-      }
+      fts.fill("pfcand_btagEtaRel", trkinfo.getTrackEtaRel());
+      fts.fill("pfcand_btagPtRatio", trkinfo.getTrackPtRatio());
+      fts.fill("pfcand_btagPParRatio", trkinfo.getTrackPParRatio());
+      fts.fill("pfcand_btagSip2dVal", trkinfo.getTrackSip2dVal());
+      fts.fill("pfcand_btagSip2dSig", trkinfo.getTrackSip2dSig());
+      fts.fill("pfcand_btagSip3dVal", trkinfo.getTrackSip3dVal());
+      fts.fill("pfcand_btagSip3dSig", trkinfo.getTrackSip3dSig());
+      fts.fill("pfcand_btagJetDistVal", trkinfo.getTrackJetDistVal());
```

Comment:
this is kind of exposing the fact that the new handling with preprocessing in the JetTags will apply catch_infs to everything.
By the time of #30072 at least this block already had catch_infs for everything here (before it was directly passed to preprocessing step in JetTags in #30072) and the transformation seemed OK.

Tests in this PR are showing no differences, but perhaps 100 events are note enough to encounter an inf/nan in the now modified variables.

To be pedantic, another flag (```bool catch_infs``` or similar) is needed (master first, then backport).
I'm not 100% sure this strictness is required though.
@hqucms @jmduarte what is the expected response to an inf/nan in the inputs made by this module to the inference taggers?

Reply:
@slava77 yes, thanks for pointing this out. I was initially worried about this, but in my own tests with ParticleNet I found 0 difference (granted it was only ~10k BulkGrav->HH->4b and ~10k ttbar events)

I assume @hqucms left this in originally for 106X because he observed some difference on some sample of events (unless he was just being extremely careful?).

Can @hqucms comment on if he expects a difference or otherwise would prefer to remove `catch_infs` from all cases?

Reply:
> @slava77 yes, thanks for pointing this out. I was initially worried about this, but in my own tests with ParticleNet I found 0 difference (granted it was only ~10k BulkGrav->HH->4b and ~10k ttbar events)

were these tests done in 106X or in the master/112X, which has/had more catch_infs already in place.

Reply:
106X.

Reply:
The `catch_infs` were not added to the `use_puppiP4_` case in https://github.com/cms-sw/cmssw/pull/29062 mainly to preserve the old behavior for DeepAK8. We observed some nan values in the DeepAK8 outputs and eventually it was tracked down to this part, so we added the `catch_infs` in the master branch and also for ParticleNet in the backport https://github.com/cms-sw/cmssw/pull/29062. 
I actually would be in favor of adding `catch_infs` for all cases if we are OK to make a small exception to the no-change policy (which only affects a very small fraction of DeepAK8 outputs that are currently nan).

Reply:
>  @hqucms @jmduarte what is the expected response to an inf/nan in the inputs made by this module to the inference taggers?

please clarify

Reply:
@slava77 I believe @hqucms is saying the previous behavior is that the tagger would output nan if any input was nan/inf. Whether that is "desirable" is another question.

In my opinion, catching the nan/infs at the input and replacing them with sensible default "null" values makes sense, as is done now in master/112X.

Here you can see the 100% agreement before and after this PR for DeepAK8 on 10k ttbar events in 106X for jet pt > 150:

http://uaf-7.t2.ucsd.edu/~woodson/restart_bound_in_producer/res_deepak8_ttbar.png
http://uaf-7.t2.ucsd.edu/~woodson/restart_bound_in_producer/corr_deepak8_ttbar.png

`file:/store/mc/RunIIFall17MiniAODv2/TTToHadronic_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/PU2017_12Apr2018_94X_mc2017_realistic_v14-v1/90000/DCFE3F5F-AE42-E811-B6DB-008CFAF72A64.root`

Reply:
> @slava77 I believe @hqucms is saying the previous behavior is that the tagger would output nan if any input was nan/inf. Whether that is "desirable" is another question.

Thanks for clarifying.
It sounds like these can be treated as a bug fix to very rare cases that could even lead to crashes downstream, which should be acceptable for the "no-change" policy.